sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/count/CountSamples.groovy

```diff
@@ -30,65 +30,30 @@ class CountSamples implements CountSamplesInput{
     this.output = output
   }
 
-  /**
-   * This method calls the output interface with the number of all samples in a project as
-   * well as the subset of samples received by a lab.
-   * In case of failure the output interface failure method is called.
-   * @since 1.0.0
-   */
   @Override
-  void countReceivedSamples(String projectCode) {
+  void countSamplePerStatus(String projectCode) {
     try {
       sampleStatuses = dataSource.fetchSampleStatusesForProject(projectCode)
+      int allSamples = sampleStatuses.size()
       // counts samples that have AT LEAST this status (or a later one)
       int receivedAmount = countSamplesFromStatus(Status.SAMPLE_RECEIVED)
-      output.countedReceivedSamples(projectCode, sampleStatuses.size(), receivedAmount)
+      output.countedReceivedSamples(projectCode,allSamples,receivedAmount)
+
+      int failedQc = sampleStatuses.findAll { it == Status.SAMPLE_QC_FAIL }.size()
+      output.countedFailedQcSamples(projectCode,allSamples,failedQc)
```

Comment:
Nit: You could extract the 2liners into small methods. This would highlight the logic more and make reading it even simpler. Plus you would actually create uncle bobs 5-10 line methods :+1:

Reply:
I like that. Will do :)